### PR TITLE
REF/CLN: pandas/io/pytables.py

### PR DESCRIPTION
**Pull request by ivanovmg**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Clean-up and minor refactor of ``pandas/io/pytables.py``.
- Extract method ``_identify_group``
- Clean-up some docstrings (spacing)
- Use suppress instead of try/except/pass
- Format error message using dedent
- Remove unnecessary empty lines as they compromise readability of some blocks

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -716,11 +712,9 @@ def open(self, mode: str = "a", **kwargs):
                     "which allows\n"
                     "files to be opened multiple times at once\n"
                 )
-
             raise err
 
         except Exception as err:
```
Comment (jbrockmendel) [inline pandas/io/pytables.py:717]: if you're up for it, narrowing down what we catch here would be really helpful

File: pandas/io/pytables.py [hunk 2]
```diff
@@ -814,19 +806,19 @@ def select(
         Parameters
         ----------
         key : str
-                Object being retrieved from file.
+            Object being retrieved from file.
         where : list, default None
-                List of Term (or convertible) objects, optional.
+            List of Term (or convertible) objects, optional.
         start : int, default None
```
Comment (jbrockmendel) [inline pandas/io/pytables.py:812]: int -> int or None

File: pandas/io/pytables.py [hunk 3]
```diff
@@ -1099,7 +1091,7 @@ def put(
             <https://pandas.pydata.org/pandas-docs/stable/user_guide/io.html#query-via-data-columns>`__.
         encoding : str, default None
             Provide an encoding for strings.
-        dropna   : bool, default False, do not write an ALL nan row to
+        dropna : bool, default False, do not write an ALL nan row to
```
Comment (jbrockmendel) [inline pandas/io/pytables.py:1094]: new line after "default False"

File: pandas/io/pytables.py [hunk 4]
```diff
@@ -1807,6 +1771,45 @@ def _read_group(self, group: "Node"):
         s.infer_axes()
         return s.read()
 
+    def _identify_group(self, key: str, append: bool) -> "Node":
+        """Identify HDF5 group based on key, delete/create group if needed."""
```
Comment (jbrockmendel) [inline pandas/io/pytables.py:1775]: """ on separate lines

File: pandas/io/pytables.py [hunk 5]
```diff
@@ -4289,9 +4275,8 @@ def write_data(self, chunksize: Optional[int], dropna: bool = False):
 
         # consolidate masks
         if len(masks):
-            mask = masks[0]
-            for m in masks[1:]:
-                mask = mask & m
+            combine_masks = lambda first, second: first & second
+            mask = reduce(combine_masks, masks)
```
Comment (jbrockmendel) [inline pandas/io/pytables.py:4279]: not a deal-breaker, but i dont find this clearer than what we have now

File: pandas/io/pytables.py [hunk 6]
```diff
@@ -5032,14 +5015,12 @@ def _maybe_adjust_name(name: str, version) -> str:
     -------
     str
     """
-    try:
+    with suppress(IndexError):
```
Comment (jbrockmendel) [inline pandas/io/pytables.py:5018]: let's do appropriate checks for this and avoid the exception altogether

File: pandas/io/pytables.py (hunk 5 above)
Comment (ivanovmg) [reply pandas/io/pytables.py:4279]: OK, I supposed it may be a questionable change. I will revert this.

File: pandas/io/pytables.py (hunk 4 above)
Comment (ivanovmg) [reply pandas/io/pytables.py:1775]: No problem, but is there a local policy for this? I mean according to PEP257 (https://www.python.org/dev/peps/pep-0257/#one-line-docstrings) if it fits in one line, then no need to put triple quotes on multiple lines.

Comment (jbrockmendel) [reply pandas/io/pytables.py:1775]: cc @datapythonista am i remembering this wrong?

Comment (datapythonista) [reply pandas/io/pytables.py:1775]: For private docstrings it doesn't make a difference, and both are correct based on Python recommendations. For public docstrings we try to standardize them to what @jbrockmendel says. We shouldn't have single liners anyway, and in our autosummaries we take the first line of the docstring (the short summary), which we want to have of similar sizes to show correctly in the html version. Having it in the line after the `"""` allows for 3 more characters, so we decided to standardize all public docstrings to that, to be consistent.

File: pandas/io/pytables.py (hunk 3 above)
Comment (ivanovmg) [reply pandas/io/pytables.py]: It turns out that this parameter is not even used in the function. Deleted the corresponding strings.

File: pandas/io/pytables.py (hunk 2 above)
Comment (ivanovmg) [reply pandas/io/pytables.py]: Changed here and in the adjacent strings.

File: pandas/io/pytables.py (hunk 6 above)
Comment (ivanovmg) [reply pandas/io/pytables.py]: I checked that the version number is a correct sequence of three elements.  I also added test as this check never hit.

File: pandas/io/pytables.py (hunk 1 above)
Comment (ivanovmg) [reply pandas/io/pytables.py:721]: I did that, looking at the source code of ``tables``. But it turns out that there was (is) no test for this exception in pandas code. So I will need to figure out how to do it, but it will require some mocking, including package version and stuff.

Comment (ivanovmg) [reply pandas/io/pytables.py:721]: Taking into account that pytables version >= 3.4.3 in current installation requirements (https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html), then I can probably remove ``ver >= 3.1`` statement.

File: pandas/io/pytables.py [hunk 7]
```diff
@@ -691,41 +697,14 @@ def open(self, mode: str = "a", **kwargs):
...
+                "Cannot open HDF5 file, which is already opened, "
+                "even in read-only mode."
+            )
+            raise ValueError(msg)
 
-            # trying to read from a non-existent file causes an error which
-            # is not part of IOError, make it one
-            if self._mode == "r" and "Unable to open/create file" in str(err):
-                raise OSError(str(err)) from err
-            raise
```
Comment (ivanovmg) [inline pandas/io/pytables.py:728]: Looks like this error message is not raised in the current version of ``tables.file``. Not even 3.1. So, I deleted this error raise.

File: pandas/io/pytables.py [hunk 8]
```diff
@@ -691,41 +697,14 @@ def open(self, mode: str = "a", **kwargs):
...
             )
 
-        try:
-            self._handle = tables.open_file(self._path, self._mode, **kwargs)
-        except OSError as err:  # pragma: no cover
-            if "can not be written" in str(err):
-                print(f"Opening {self._path} in read-only mode")
-                self._handle = tables.open_file(self._path, "r", **kwargs)
-            else:
-                raise
```
Comment (ivanovmg) [inline pandas/io/pytables.py:701]: I checked the current code of ``tables.open_file`` and did not find error messages containing ``can not be written``. Even in 3.1 there is no such message. Considering the current minimal pytables version (>= 3.4.3), this error handing seems to be irrelevant (``if "can not be written" in str(err)``).

Comment (jreback) [reply pandas/io/pytables.py:701]: yeah this could have changed, was hard to test :->

File: pandas/io/pytables.py [hunk 9]
```diff
@@ -1735,38 +1707,12 @@ def _write_to_group(
...
     ):
-        group = self.get_node(key)
-
-        # we make this assertion for mypy; the get_node call will already
-        #  have raised if this is incorrect
-        assert self._handle is not None
-
-        # remove the node if we are not appending
-        if group is not None and not append:
-            self._handle.remove_node(group, recursive=True)
```
Comment (jreback) [inline pandas/io/pytables.py:1746]: what happened to this?

**Review by jreback — CHANGES_REQUESTED** [review]
looks fine, just a question.

File: pandas/io/pytables.py (hunk 9 above)
Comment (ivanovmg) [reply pandas/io/pytables.py:1746]: I put it in the method ``_identify_group`` below.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ivanovmg
